packages/react-components/src/util/observer.tsx

```diff
@@ -1,36 +1,29 @@
-import React, { PureComponent, ReactNode } from 'react'
+import React, { FC, ReactNode, useEffect, useRef } from 'react'
 
 type Props = {
+    // can't get FC<Props> to pass eslintreact/prop-types warning :/
     children: ReactNode
     onVisibleChange: (isVisible: boolean) => void
     className?: string
 }
 
-class Observer extends PureComponent<Props> {
-    io: any
-    container?: HTMLDivElement | null
-    componentDidMount() {
-        this.io = new IntersectionObserver(([entry]: IntersectionObserverEntry[]) => {
-            const { onVisibleChange } = this.props
-            if (onVisibleChange) onVisibleChange(entry.isIntersecting)
-        })
-        this.io.observe(this.container)
-    }
-
-    componentWillUnmount() {
-        if (this.io) {
-            this.io.disconnect()
+const Observer: FC<Props> = ({ children, className, onVisibleChange }: Props) => {
```

Comment:
This was just converted to a hook because I though I needed to check `document.visibilityState`, and was gonna do that in a hook